lib/elastic_record/index.rb

```diff
@@ -25,30 +23,30 @@ module ElasticRecord
   #   Returns if the index exists
   # [get_mapping]
   #   Returns the mapping currently stored by elastic search.
-  # [put_mapping]
+  # [update_mapping]
   #   Update elastic search's mapping
   class Index
     include Documents
     include Manage
     include Mapping, Settings
-    include Percolator
     include Analyze
     include Deferred
 
-    attr_accessor :model
+    attr_accessor :doctypes
 
     attr_accessor :disabled
-    attr_accessor :has_percolator
+    attr_accessor :model
     attr_accessor :partial_updates
 
-    def initialize(model)
-      @model = model
+    def initialize(models)
+      models = Array.wrap(models)
+      @model = models.first
+      @doctypes = models.map(&:doctype)
       @disabled = false
     end
```

Comment:
The support for multiple doctypes, alongside the arbitrary selection of the first model, is disgruntling. Is the plural support necessary?